app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationsFragment.kt

```diff
@@ -366,24 +359,19 @@ class NotificationsFragment :
                 }
 
                 // Update status display from statusDisplayOptions. If the new options request
-                // relative time display collect the flow to periodically re-bind the UI.
+                // relative time display collect the flow to periodically update the timestamp in the list gui elements.
                 launch {
                     viewModel.statusDisplayOptions
                         .collectLatest {
+                            // NOTE this this also triggered (emitted?) on resume.
+
                             adapter.statusDisplayOptions = it
-                            layoutManager.findFirstVisibleItemPosition().let { first ->
-                                first == RecyclerView.NO_POSITION && return@let
-                                val count = layoutManager.findLastVisibleItemPosition() - first
-                                adapter.notifyItemRangeChanged(
-                                    first,
-                                    count,
-                                    null
-                                )
-                            }
+                            adapter.notifyItemRangeChanged(0, adapter.itemCount, null)
 
                             if (!it.useAbsoluteTime) {
                                 updateTimestampFlow.collect()
                             }
+                            // TODO there is a way to stop this?
```

Comment:
If I understand this code correctly (collectLatest) it should always do something on changes to the display options.

So also (only) "useAbsoluteTime" option could change and if it is set to true the flow is unnecessary.

However currently this does not seem to be a problem as there is a sort of full restart on display preferences change (you will always return to the home timeline which is reloaded).